debezium-core/src/main/java/io/debezium/schema/AbstractUnicodeTopicNamingStrategy.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.schema;
+
+import java.util.Properties;
+
+import io.debezium.common.annotation.Incubating;
+import io.debezium.spi.schema.DataCollectionId;
+import io.debezium.spi.topic.TopicNamingStrategy;
+
+/**
+ * An abstract unicode converter topic naming strategy implementation of {@link TopicNamingStrategy}.
+ * Any non encodable character or underscore would be encoded via _uxxxx sequence in the same way as Java works,
+ * and _ is an escape sequence like backslash in Java.
+ *
+ * @author Harvey Yue
+ */
+@Incubating
+public abstract class AbstractUnicodeTopicNamingStrategy extends AbstractTopicNamingStrategy<DataCollectionId> {
+
+    private static final ReplacementFunction UNICODE_REPLACEMENT = c -> {
+        String hex = Integer.toHexString(c);
+        if (hex.length() <= 2) {
+            hex = "00" + hex;
+        }
+        return "_u" + hex;
+    };
+
+    public AbstractUnicodeTopicNamingStrategy(Properties props) {
+        super(props);
+    }
+
+    @Override
+    public String sanitizedTopicName(String topicName) {
+        return sanitizedTopicName(topicName, UNICODE_REPLACEMENT);
+    }
+
+    @Override
+    public boolean isValidKafkaTopicCharacter(char c) {
```

Comment:
The function name is not clear. What is valid Kafka topic character is defined by Kafka itself and we reduce the set here.